stage1/init/common/mount.go

```diff
@@ -238,3 +242,106 @@ func ensureDestinationExists(source, destination string) error {
 	}
 	return nil
 }
+
+func AppAddMounts(p *stage1commontypes.Pod, ra *schema.RuntimeApp, enterCmd []string) {
+	vols := make(map[types.ACName]types.Volume)
+	for _, v := range p.Manifest.Volumes {
+		vols[v.Name] = v
+	}
+
+	imageManifest := p.Images[ra.Name.String()]
+
+	/* TODO(alban): ra.Mounts is empty at the moment because the run-time manifest
+	 * has not been updated with the mount. See how it is done in
+	 * generatePodManifest/MergeMounts.
+	 *
+	 * For now, do it manually.
+	 */
+	ra.Mounts = append(ra.Mounts, schema.Mount{Volume: "foo", Path: "/mnt/hardcoded"})
+
+	mounts := GenerateMounts(ra, vols, imageManifest)
+	for _, m := range mounts {
+		vol := vols[m.Volume]
+		AppAddOneMount(p, ra, vol.Source, m.Mount.Path, enterCmd)
+	}
+}
+
+func AppAddOneMount(p *stage1commontypes.Pod, ra *schema.RuntimeApp, sourcePath string, dstPath string, enterCmd []string) {
```

Comment:
This is mirroring the algorithm from [bus_machine_method_bind_mount](https://github.com/systemd/systemd/blob/v231/src/machine/machine-dbus.c#L865) in systemd.

You're right, I will add comments.